nupic/research/frameworks/continuous_learning/dendrite_layers.py

```diff
@@ -0,0 +1,153 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2020, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import numpy as np
+import torch
+import torch.nn.functional as F
+from torch import nn
+
+from dend_kwinners import DKWinners
+from nupic.torch.modules import SparseWeights  # , KWinnersBase, KWinners
+
+
+class DendriteInput(nn.Module):
+    def __init__(self,
```

Comment:
Docstrings here would help. The level of documentation required increases with the importance of the function - if this is the key to the whole thing, might be good to add a few examples of expected behavior as well.